crates/lib3h/src/gateway/gateway_dht.rs

```diff
@@ -102,7 +102,7 @@ impl<T: Transport, D: Dht> P2pGateway<T, D> {
                     // If no connection to that transportId is open, open one first.
                     self.inner_transport
                         .borrow_mut()
-                        .send(&[peer_transport.as_str()], &payload)?;
+                        .send(&[peer_transport.path()], &payload)?;
```

Comment:
This should be wrapped in the `transport::url_to_transport_id` function so we can easily isolate the subtle assumptions later. I will do it in a future PR.